pkg/schemapatcher/gen.go

```diff
@@ -125,92 +137,149 @@ func (g Generator) Generate(ctx *genall.GenerationContext) (result error) {
 				fullSchema = *fullSchema.DeepCopy()
 				crdgen.TruncateDescription(&fullSchema, *g.MaxDescLen)
 			}
-			existingInfo.NewSchemata[gv.Version] = fullSchema
+			existingSet.NewSchemata[gv.Version] = fullSchema
 		}
 	}
 
 	// patch existing CRDs with new schemata
-	for _, existingInfo := range partialCRDs {
+	for _, existingSet := range partialCRDSets {
 		// first, figure out if we need to merge schemata together if they're *all*
 		// identical (meaning we also don't have any "unset" versions)
 
-		if len(existingInfo.NewSchemata) == 0 {
+		if len(existingSet.NewSchemata) == 0 {
 			continue
 		}
 
 		// copy over the new versions that we have, keeping old versions so
-		// that we can tell if a schema would be nill
+		// that we can tell if a schema would be nil
 		var someVer string
-		for ver := range existingInfo.NewSchemata {
+		for ver := range existingSet.NewSchemata {
 			someVer = ver
-			existingInfo.Versions[ver] = struct{}{}
+			existingSet.Versions[ver] = struct{}{}
 		}
 
 		allSame := true
-		firstSchema := existingInfo.NewSchemata[someVer]
-		for ver := range existingInfo.Versions {
-			otherSchema, hasSchema := existingInfo.NewSchemata[ver]
+		firstSchema := existingSet.NewSchemata[someVer]
+		for ver := range existingSet.Versions {
+			otherSchema, hasSchema := existingSet.NewSchemata[ver]
 			if !hasSchema || !equality.Semantic.DeepEqual(firstSchema, otherSchema) {
 				allSame = false
 				break
 			}
 		}
 
 		if allSame {
-			if err := existingInfo.setGlobalSchema(); err != nil {
-				return fmt.Errorf("failed to set global firstSchema for %s: %v", existingInfo.GroupKind, err)
+			if err := existingSet.setGlobalSchema(); err != nil {
+				return fmt.Errorf("failed to set global firstSchema for %s: %v", existingSet.GroupKind, err)
 			}
 		} else {
-			if err := existingInfo.setVersionedSchemata(); err != nil {
-				return fmt.Errorf("failed to set versioned schemas for %s: %v", existingInfo.GroupKind, err)
+			if err := existingSet.setVersionedSchemata(); err != nil {
+				return fmt.Errorf("failed to set versioned schemas for %s: %v", existingSet.GroupKind, err)
 			}
```

Comment:
the top-level field does not exist anymore in v1. You always set schemas, subresources etc. in the versions.

Reply:
yep.  I can probably make it clearer, but if you look at the source for `setGlobalSchema`, it only does so if it's looking at a v1beta1 CRD, and delegates back to `setVersionedSchemata` for the v1 CRD.  We might want to rename the methods...

Reply:
yep, looks good